e2e_tests/tests/command/test_tensorboard.py

```diff
@@ -159,3 +160,44 @@ def test_start_tensorboard_for_multi_experiment(tmp_path: Path, secrets: Dict[st
                 break
         else:
             raise AssertionError(f"Did not find {SERVICE_READY} in output")
+
+
+@pytest.mark.slow
+@pytest.mark.e2e_gpu
+@pytest.mark.tensorflow2
+@pytest.mark.parametrize("prefix", [None, "my/test/prefix"])
+def test_start_tensorboard_with_custom_image(
+    tmp_path: Path, secrets: Dict[str, str], prefix: Optional[str]
+) -> None:
+    """
+    Start a random experiment configured with the shared_fs backend, start a
+    TensorBoard instance pointed to the experiment, and kill the TensorBoard
+    instance.
+    """
+    with FileTree(tmp_path, {"config.yaml": s3_config(1, secrets, prefix)}) as tree:
+        config_path = tree.joinpath("config.yaml")
+        experiment_id = exp.run_basic_test(
+            str(config_path), conf.fixtures_path("no_op"), num_trials
+        )
+
+    command = [
+        "tensorboard",
+        "start",
+        str(experiment_id),
+        "--no-browser",
+        "--detach",
+        "--config",
+        f"environment.image={custom_image_for_testing}",
+    ]
+    with cmd.interactive_command(*command) as tensorboard:
+        t_id = tensorboard.task_id
+        commandt = ["tensorboard", "config", t_id]
+        with cmd.interactive_command(*commandt, task_id=t_id) as tensorboard_config:
```

Comment:
no need for this to be interactive either; you could pass the entire output of the command into a yaml parser and check the output that way, which would be clearer to read as well.